File: mobile/ios/SpeakeasyCrypto/SignalStores.swift
import Foundation
// Placeholder for LibSignal imports if available via SPM
// import SignalProtocol

// Defines the storage requirements for Speakeasy Crypto

protocol SpeakeasyStore {
    // Identity Key Storage
    func getIdentityKeyPair() -> Data? // Returns Serialized KeyPair
    func getLocalRegistrationId() -> UInt32?
    func saveIdentity(_ keyPair: Data, registrationId: UInt32)
    
    // PreKey Storage
    func loadPreKey(id: UInt32) -> Data?
    func storePreKey(_ key: Data, id: UInt32)
    func containsPreKey(id: UInt32) -> Bool
    func removePreKey(id: UInt32)
    
    // Signed PreKey Storage
    func loadSignedPreKey(id: UInt32) -> Data?
    func storeSignedPreKey(_ key: Data, id: UInt32)
    func containsSignedPreKey(id: UInt32) -> Bool
    func removeSignedPreKey(id: UInt32)
    
    // Session Storage
    func loadSession(address: String) -> Data? // Address = "UserId.DeviceId"
    func storeSession(_ session: Data, address: String)
    func containsSession(address: String) -> Bool
    func deleteSession(address: String)
}

File: mobile/android/crypto/SignalStores.kt
package com.speakeasy.crypto

// Defines the storage requirements for Speakeasy Crypto

interface SpeakeasyStore {
    // Identity Key Storage
    fun getIdentityKeyPair(): ByteArray? // Returns Serialized KeyPair
    fun getLocalRegistrationId(): Int?
    fun saveIdentity(keyPair: ByteArray, registrationId: Int)
    
    // PreKey Storage
    fun loadPreKey(id: Int): ByteArray?
    fun storePreKey(key: ByteArray, id: Int)
    fun containsPreKey(id: Int): Boolean
    fun removePreKey(id: Int)
    
    // Signed PreKey Storage
    fun loadSignedPreKey(id: Int): ByteArray?
    fun storeSignedPreKey(key: ByteArray, id: Int)
    fun containsSignedPreKey(id: Int): Boolean
    fun removeSignedPreKey(id: Int)
    
    // Session Storage
    fun loadSession(address: String): ByteArray? // Address = "UserId.DeviceId"
    fun storeSession(session: ByteArray, address: String)
    fun containsSession(address: String): Boolean
    fun deleteSession(address: String)
}
